server/Arcadia.Assistant.ExternalStorages.Abstractions/IExternalStorage.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Arcadia.Assistant.ExternalStorages.Abstractions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public interface IExternalStorage : IDisposable
+    {
+        Task<IEnumerable<StorageItem>> GetItems(string list, IEnumerable<ICondition> conditions = null, CancellationToken cancellationToken = default(CancellationToken));
+
+        Task<StorageItem> AddItem(string list, StorageItem item, CancellationToken cancellationToken = default(CancellationToken));
+
+        Task UpdateItem(string list, string itemId, StorageItem item, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
doesn't StorageItem contain an id?

Reply:
It has, but it is a bit less intuitive to imply that storage should take one part of object to get an item and update the other part of object in it, as for me.